### PR TITLE
Python UFC SDK updates

### PR DESCRIPTION
**Pull request by schmit**
---
labels: mergeable
---
[//]: #  (Link to the issue corresponding to this chunk of work)
Fixes: #__issue__

## Motivation and Context
[//]: #  (Why is this change required? What problem does it solve?)

## Description
[//]: # (Describe your changes in detail)

## How has this been tested?
[//]: # (Please describe in detail how you tested your changes)

[//]: # (OPTIONAL)
[//]: # (Add one or multiple labels: enhancement, refactoring, bugfix)

### CODE REVIEW

File: eppo_client/client.py [hunk 1]
```diff
@@ -74,12 +74,15 @@ def get_numeric_assignment(
         default: float,
         subject_attributes: Optional[SubjectAttributes] = None,
     ) -> float:
-        return self.get_assignment_variation(
-            subject_key,
-            flag_key,
-            default,
-            subject_attributes,
-            VariationType.NUMERIC,
+        # convert to float in case we get an int
```
Comment (giorgiomartini0) [inline eppo_client/client.py:77]: 👍  Took me a moment to understand – I first thought you were going to rescue a mismatch (`getNumeric` for an `Integer` flag) which I didn't like. But what you're doing is ensuring that `3.0` (a valid variation for a numeric flag) is indeed returned as a float and not as the integer `3`.

File: eppo_client/client.py [hunk 2]
```diff
@@ -227,3 +236,20 @@ def check_type_match(
...
+) -> bool:
+    if expected_type is None:
+        return True
+    if expected_type in [VariationType.JSON, VariationType.STRING]:
+        return isinstance(value, str)
+    if expected_type == VariationType.INTEGER:
+        return isinstance(value, int)
+    if expected_type == VariationType.NUMERIC:
+        # we can convert int to float
+        return isinstance(value, float) or isinstance(value, int)
```
Comment (giorgiomartini0) [inline eppo_client/client.py:252]: `or isinstance(value, int)` seems unnecessary? With this PR we _always_ convert to float? Other than in one of the tests where I see that you explicitly rely on this behavior. In any case this is fine to keep - it's more that I want to check my understanding.

File: eppo_client/rules.py [hunk 3]
```diff
@@ -40,6 +41,11 @@ def evaluate_condition(
     condition: Condition, subject_attributes: SubjectAttributes
 ) -> bool:
     subject_value = subject_attributes.get(condition.attribute, None)
+    if condition.operator == OperatorType.IS_NULL:
+        if condition.value:
+            return subject_value is None
+        return subject_value is not None
```
Comment (giorgiomartini0) [inline eppo_client/rules.py:47]: supernit, optional: could write as `return (subject_value is None) if condition.value else (subject_value is not None)`

**Review by giorgiomartini0 — APPROVED** [review]
Nothing blocking

File: eppo_client/client.py (hunk 2 above)
Comment (schmit) [reply eppo_client/client.py:252]: Without the `or` this check would fail for an integer value such as `1`, which we can convert to a float

Comment (giorgiomartini0) [reply eppo_client/client.py:255]: I get it now. The cast to `float` is done _after_ validating the value type, so this is indeed necessary.

**Pull request merged**